src/plugins/ui_core_views/pivot_ui.ts

```diff
@@ -65,9 +66,7 @@ export class PivotUIPlugin extends UIPlugin {
   handle(cmd: Command) {
     if (invalidateEvaluationCommands.has(cmd.type)) {
       for (const pivotId of this.getters.getPivotIds()) {
-        if (!pivotRegistry.get(this.getters.getPivotCoreDefinition(pivotId).type).externalData) {
```

Comment:
If I understand correctly, there's no bug in o-spreadsheet (all tests in this PR are :heavy_check_mark: in without the diff in the code), but this is this line which prevents from updating odoo pivots ? and also the deepcopy

But then, are the tests introduced/changed here useful ?